### DISCUSSION TITLE
Directus post to collection that stores multiple image files

### DISCUSSION THREAD

**Discussion by Mbengparis** [directus/directus#5077] [Archive]
I am trying to make a POST request to add an item to my collection. 

My collection has a title, description and an image field ( multiple files).
Here is an example of my code:

<--Html code-->

form:{
        title: '',
        description: ''
        images: []
      } 

const formData = new FormData();

      formData.append("title", this.form.title);
      formData.append("description", this.form.description);
      formData.append("images", this.form.images);

axios.post('http://localhost/items/collection/?access_token=token', formData).then()

The above post request stores empty fields into my collection. 

I have some questions:::::

1) Is it possible for images to be stored in a field in the collection and at same time in file ?. 
If yes thats what i want to achieve.

Please i need help. What am i doing wrong ?

**Comment by rijkvanzanten** [comment]
You can't use multipart/form-data to submit to a regular item. Upload the files first using the `POST /files` endpoint, and save the resulting UUIDs under `images` in the  `POST /items/collection`

**Reply by Mbengparis** [reply]
I did exactly as you have said and i get an error 403 vorbidden.  The image gets saved to files in directus and i get the image response and saved the id to this.form.images. Please what can be the problem.

In directus my field is a multiple file field, meaning i can save either one or many images. But when i do manual upload in directus, everything works perfect which menas its not a permission problem i guess.

Here is an example of what you said:

```
<--Html code-->

form: {
title: '',
description: ''
images: []
}
```

```js
const formData = new FormData();

formData.append('file', image);

 axios.post(`http://localhost/files/?access_token=${token}`, formData, 
  {
    headers:{
    'Content-Type': 'multipart/form-data; charset=utf-8; boundary=__X_BOUNDARY__',
  }
  })
.then((res) =>{
 
/// on response I pushed the image id from my files to my form image

this.form.images.push(res.data.data.id)
 
   
  axios.post(`http://localhost/items/collection?access_token=${token}`, this.form) 

/// Here on response i get 403 error
}
```

**Reply by rijkvanzanten** [reply]
That 403 could be that your current user doesn't have access to the `collection` collection, or that collection or one of the fields in it doesn't exist. Is the collection you're saving to called `collection`? Do those three fields exist? Is your user authorized to create items in this collection?

**Comment by Mbengparis** [comment]
I just put collection for explanation purposes. It's actually called Work. All of the fields I mentioned above do exist in my collection (Work). 
Yes my user is authorized to create items.  You check my roles and permission for the user. 
For the custom fields I just put {"user_created": {"_eq": "$CURRENT_USER"}}
Thanks in advance.

**Reply by rijkvanzanten** [reply]
@Mbengparis Try it out with an admin user first to rule out any permissions issues. If that works, then double check your permissions configuration to make sure your user has access. If it doesn't work as an admin user, make sure you got all the names correct, and are sending the correct post body 👍🏻

**Comment by Mbengparis** [comment]
I tried it as an admin user and it's same error. I also checked the field names they are all correct. 
I think it should be the post body. 
The thing is if I post commenting out the image field the other fields gets posted to the backend with no error. 

I realized that the image file is saved in the collection as an interger but my UUID is in the following CHAR format:

"id": "4f4b14fa-a43a-46d0-b7ad-90af5919bebb"

If i push into the image field an integer for example to test, I don't get any error but if I push in the above UUID  I get the error because the types are different. 
I think this is where the problem lies. 

File saves images with id of CHAR type whereas the image field saves images with type INT.

Is it possible that when saving into the files, the UUID should be an int Typ  ? 


Am Freitag, April 16, 2021, 10:20 PM schrieb Rijk van Zanten ***@***.***>:




@Mbengparis Try it out with an admin user first to rule out any permissions issues. If that works, then double check your permissions configuration to make sure your user has access. If it doesn't work as an admin user, make sure you got all the names correct, and are sending the correct post body 👍🏻

**Comment by Mbengparis** [comment]
Thanks i got it finally working. The probelm is i was supposed to post UUID above directly to the directus_files_id as such:

this.form.images.push({"directus_files_id": res.data.data.id})

Its working fine now. Thanks for the support
